### ISSUE TITLE
TypeError: "model is undefined"

### ISSUE THREAD

**Issue by alessio265** [linkedpipes/etl#572]
Hi, I'm also a PhD student and I'm really interested in using LP-ETL. But after installing it on Ubuntu, when I'm trying to insert a new component in LP-ETL, I encounter a TypeError: "model is undefined" in Firefox 61.0.1 Web Developper, making LP-ETL unfortunately unusable... any ideas what I'm missing? Thanks

**Comment by skodapetr** [comment]
Hi, can I ask are you using develop or a master branch?

**Comment by jakubklimek** [comment]
@alessio265 Please try https://demo.etl.linkedpipes.com - it is our demo instance with the latest develop branch deployed. If this works, it is probably some problem with configuration. Did you make any changes to configuration.properties?

**Comment by alessio265** [comment]
Thanks for your quick answer! 👍 
@skodapetr It is master branch. 
@jakubklimek didn't change anything on the configuration, just copied the sample as explained in the doc --> cp configuration.properties.sample deploy/configuration.properties

**Comment by jakubklimek** [comment]
@alessio265 OK, we will investigate. Does the demo instance work? If so, could you try the develop branch?

**Comment by alessio265** [comment]
@jakubklimek yes, the demo instance works. I just tried with the develop branch and I have the same issue, I have the following error messages  "can't load pipeline, model is undefined" and "can't fetch pipeline".

**Comment by jakubklimek** [comment]
@alessio265 and, are all 4 components running, ie. executor, executor-monitor, storage and frontend?

**Comment by alessio265** [comment]
@jakubklimek Yes, it is http://localhost:8080 and yes all 4 components are running. I'm suspecting an issue with storage component as it is throwing com.linkedpipes.etl.storage.BaseException,java.io.IOException

**Comment by jakubklimek** [comment]
Coud you include a more complete stack trace of the exception?

**Comment by alessio265** [comment]
16:32:36,639 INFO  c.l.e.s.Configuration - Reading configuration file: configuration.properties
16:32:37,208 INFO  o.s.c.s.ClassPathXmlApplicationContext - Refreshing org.springframework.context.support.ClassPathXmlApplicationContext@65e98b1c: startup date [Mon Aug 13 16:32:37 CEST 2018]; root of context hierarchy
16:32:37,369 INFO  o.s.b.f.x.XmlBeanDefinitionReader - Loading XML bean definitions from class path resource [spring/context-storage.xml]
16:32:39,068 INFO  o.s.s.c.ThreadPoolTaskExecutor - Initializing ExecutorService 
16:32:39,078 INFO  o.s.c.s.PostProcessorRegistrationDelegate$BeanPostProcessorChecker - Bean 'asynchExecutor' of type [class org.springframework.scheduling.config.TaskExecutorFactoryBean] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
16:32:39,080 INFO  o.s.c.s.PostProcessorRegistrationDelegate$BeanPostProcessorChecker - Bean 'asynchExecutor' of type [class org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
16:32:39,203 INFO  c.l.e.s.Configuration - Reading configuration file: configuration.properties
16:32:39,265 INFO  c.l.e.s.jar.JarManager - Loading JAR components ...
16:32:40,954 INFO  c.l.e.s.jar.JarManager - Loading JAR components ... done
16:32:43,384 INFO  c.l.e.s.web.WebServer - Starting server on port: 8083
16:32:43,457 INFO  o.eclipse.jetty.util.log - Logging initialized @8163ms to org.eclipse.jetty.util.log.Slf4jLog
16:32:43,739 INFO  o.e.jetty.server.Server - jetty-9.4.11.v20180605; built: 2018-06-05T18:24:03.829Z; git: d5fc0523cfa96bfebfbda19606cad384d772f04c; jvm 10.0.2+13
16:32:43,870 INFO  o.e.j.s.h.C.ROOT - Initializing Spring root WebApplicationContext
16:32:43,873 INFO  o.s.w.c.ContextLoader - Root WebApplicationContext: initialization started
16:32:43,883 INFO  o.s.w.c.s.XmlWebApplicationContext - Refreshing Root WebApplicationContext: startup date [Mon Aug 13 16:32:43 CEST 2018]; parent: org.springframework.context.support.ClassPathXmlApplicationContext@65e98b1c
16:32:43,890 INFO  o.s.b.f.x.XmlBeanDefinitionReader - Loading XML bean definitions from ServletContext resource [/spring/context-web.xml]
16:32:44,639 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/pipelines/unpack],methods=[POST]}" onto public void com.linkedpipes.etl.storage.web.servlet.PipelineServlet.unpackPipeline(java.lang.String,org.springframework.web.multipart.MultipartFile,org.springframework.web.multipart.MultipartFile,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws com.linkedpipes.etl.storage.BaseException
16:32:44,642 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/pipelines/info],methods=[GET]}" onto public void com.linkedpipes.etl.storage.web.servlet.PipelineServlet.getDesignInformation(javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws com.linkedpipes.etl.storage.BaseException
16:32:44,642 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/pipelines/list],methods=[GET]}" onto public void com.linkedpipes.etl.storage.web.servlet.PipelineServlet.getPipelines(javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws com.linkedpipes.etl.storage.BaseException
16:32:44,643 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/pipelines],methods=[POST],consumes=[multipart/form-data]}" onto public void com.linkedpipes.etl.storage.web.servlet.PipelineServlet.createPipeline(org.springframework.web.multipart.MultipartFile,org.springframework.web.multipart.MultipartFile,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws com.linkedpipes.etl.storage.BaseException
16:32:44,643 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/pipelines],methods=[PUT],consumes=[multipart/form-data]}" onto public void com.linkedpipes.etl.storage.web.servlet.PipelineServlet.updatePipeline(java.lang.String,org.springframework.web.multipart.MultipartFile,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws com.linkedpipes.etl.storage.BaseException
16:32:44,646 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/pipelines],methods=[DELETE]}" onto public void com.linkedpipes.etl.storage.web.servlet.PipelineServlet.deletePipeline(java.lang.String,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws com.linkedpipes.etl.storage.BaseException
16:32:44,646 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/pipelines/localize],methods=[POST],consumes=[multipart/form-data]}" onto public void com.linkedpipes.etl.storage.web.servlet.PipelineServlet.localizePipeline(org.springframework.web.multipart.MultipartFile,org.springframework.web.multipart.MultipartFile,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws com.linkedpipes.etl.storage.BaseException
16:32:44,647 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/pipelines],methods=[GET]}" onto public void com.linkedpipes.etl.storage.web.servlet.PipelineServlet.getPipeline(java.lang.String,boolean,boolean,boolean,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws com.linkedpipes.etl.storage.BaseException
16:32:44,651 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/jars/file],methods=[GET]}" onto public void com.linkedpipes.etl.storage.web.servlet.JarServlet.getJarFile(java.lang.String,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws java.io.IOException
16:32:44,733 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/components/config],methods=[GET]}" onto public void com.linkedpipes.etl.storage.web.servlet.ComponentServlet.getConfig(java.lang.String,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws java.io.IOException,com.linkedpipes.etl.storage.BaseException
16:32:44,739 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/components/interface],methods=[GET]}" onto public void com.linkedpipes.etl.storage.web.servlet.ComponentServlet.getComponentInterface(java.lang.String,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws java.io.IOException,com.linkedpipes.etl.storage.BaseException
16:32:44,747 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/components/component],methods=[POST]}" onto public void com.linkedpipes.etl.storage.web.servlet.ComponentServlet.updateComponent(java.lang.String,org.springframework.web.multipart.MultipartFile,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws java.io.IOException,com.linkedpipes.etl.storage.BaseException
16:32:44,747 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/components/configEffective],methods=[GET]}" onto public void com.linkedpipes.etl.storage.web.servlet.ComponentServlet.getEffectiveConfig(java.lang.String,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws java.io.IOException,com.linkedpipes.etl.storage.BaseException
16:32:44,748 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/components/configTemplate],methods=[GET]}" onto public void com.linkedpipes.etl.storage.web.servlet.ComponentServlet.getConfigTemplate(java.lang.String,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws java.io.IOException,com.linkedpipes.etl.storage.BaseException
16:32:44,749 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/components],methods=[POST],consumes=[multipart/form-data]}" onto public void com.linkedpipes.etl.storage.web.servlet.ComponentServlet.createComponent(org.springframework.web.multipart.MultipartFile,org.springframework.web.multipart.MultipartFile,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws com.linkedpipes.etl.storage.BaseException,java.io.IOException
16:32:44,754 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/components/usage],methods=[GET]}" onto public void com.linkedpipes.etl.storage.web.servlet.ComponentServlet.getUsage(java.lang.String,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws com.linkedpipes.etl.storage.BaseException,java.io.IOException
16:32:44,758 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/components/config],methods=[POST]}" onto public void com.linkedpipes.etl.storage.web.servlet.ComponentServlet.updateConfig(java.lang.String,org.springframework.web.multipart.MultipartFile,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws java.io.IOException,com.linkedpipes.etl.storage.BaseException
16:32:44,764 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/components/dialog],methods=[GET]}" onto public void com.linkedpipes.etl.storage.web.servlet.ComponentServlet.getDialogResource(java.lang.String,java.lang.String,java.lang.String,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws java.io.IOException
16:32:44,765 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/components/static],methods=[GET]}" onto public void com.linkedpipes.etl.storage.web.servlet.ComponentServlet.getStaticResource(java.lang.String,java.lang.String,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws java.io.IOException
16:32:44,777 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/components/configDescription],methods=[GET]}" onto public void com.linkedpipes.etl.storage.web.servlet.ComponentServlet.getConfigDescription(java.lang.String,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws java.io.IOException,com.linkedpipes.etl.storage.BaseException
16:32:44,777 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/components/list],methods=[GET]}" onto public void com.linkedpipes.etl.storage.web.servlet.ComponentServlet.getAll(javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws java.io.IOException,com.linkedpipes.etl.storage.BaseException
16:32:44,778 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/components],methods=[DELETE]}" onto public void com.linkedpipes.etl.storage.web.servlet.ComponentServlet.remove(java.lang.String,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws java.io.IOException,com.linkedpipes.etl.storage.BaseException
16:32:44,778 INFO  o.s.w.s.m.m.a.RequestMappingHandlerMapping - Mapped "{[/components/definition],methods=[GET]}" onto public void com.linkedpipes.etl.storage.web.servlet.ComponentServlet.getDefinition(java.lang.String,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse) throws java.io.IOException,com.linkedpipes.etl.storage.BaseException
16:32:45,389 INFO  o.s.w.s.m.m.a.RequestMappingHandlerAdapter - Looking for @ControllerAdvice: Root WebApplicationContext: startup date [Mon Aug 13 16:32:43 CEST 2018]; parent: org.springframework.context.support.ClassPathXmlApplicationContext@65e98b1c
16:32:45,568 INFO  o.s.w.s.m.m.a.RequestMappingHandlerAdapter - Looking for @ControllerAdvice: Root WebApplicationContext: startup date [Mon Aug 13 16:32:43 CEST 2018]; parent: org.springframework.context.support.ClassPathXmlApplicationContext@65e98b1c
16:32:45,749 INFO  o.s.w.s.m.m.a.RequestMappingHandlerAdapter - Looking for @ControllerAdvice: Root WebApplicationContext: startup date [Mon Aug 13 16:32:43 CEST 2018]; parent: org.springframework.context.support.ClassPathXmlApplicationContext@65e98b1c
16:32:45,902 INFO  o.s.w.c.ContextLoader - Root WebApplicationContext: initialization completed in 2029 ms
16:32:45,907 INFO  o.e.j.s.h.C.ROOT - Initializing Spring FrameworkServlet 'org.springframework.web.servlet.DispatcherServlet-1cec219f'
16:32:45,908 INFO  o.s.w.s.DispatcherServlet - FrameworkServlet 'org.springframework.web.servlet.DispatcherServlet-1cec219f': initialization started
16:32:46,007 INFO  o.s.w.s.DispatcherServlet - FrameworkServlet 'org.springframework.web.servlet.DispatcherServlet-1cec219f': initialization completed in 98 ms
16:32:46,011 INFO  o.e.j.s.h.ContextHandler - Started o.e.j.s.ServletContextHandler@131ff6fa{/,jar:file:/home/alessio/Downloads/linkedpipes/etl/deploy/storage/storage.jar!/web/,AVAILABLE}
16:32:46,181 INFO  o.e.j.s.AbstractConnector - Started ServerConnector@3f725306{HTTP/1.1,[http/1.1]}{0.0.0.0:8083}
16:32:46,182 INFO  o.e.jetty.server.Server - Started @10888ms

**Comment by jakubklimek** [comment]
@alessio265  That is actually not an exception, just a web service binding to Java methods. Could you see what are the errors in your Firefox Dev Console?

**Comment by jakubklimek** [comment]
@alessio265 I just tried installing on a fresh Ubuntu like this (tell me where you did something different)

1. Ubuntu 18.04.1, Oracle Java 1.8.0_171, Apache Maven 3.5.2, Node.js 10.8.0
2. `git clone https://github.com/linkedpipes/etl.git`
3. `cd etl`
4. `mvn install`
5. `cd deploy` (here you copies `configuration.properties.sample` which is however no longer necessary
6. `./executor.sh >> executor.log &`
7. `./executor-monitor.sh >> executor-monitor.log &`
8. ` ./storage.sh >> storage.log &`
9. `./frontend.sh >> frontend.log &`
10. Access `localhost:8080`
11. Create pipeline
12. Click in canvas, add component, select component from the list, see the component
13. Click on `Execute`
14. The pipeline runs

**Comment by alessio265** [comment]
@jakubklimek 
I restarded my virtual machine and started all over from point 6. and now it is working. 
I don't know what was wrong, but here are the steps I followed :
1. I tried with java 8 and java 10, my version of Node.js is 8.11.3, all my other settings are equal to yours
2-5. Exactly, what I have done.
6-9. Yes I run all in one row, without logging.
10. Yes, it works!
11. When I wanted to create a pipeline (just after the page loaded) your application returned "can't load pipeline, model is undefined" and "can't fetch pipeline" and Firefox "model is undefined".
12-14. Impossible to add a components.

Thank you for your quick reaction and investment.
